yt_dlp/utils.py

```diff
@@ -5306,6 +5306,8 @@ def _traverse_obj(obj, path, _current_depth=0):
         for i, key in enumerate(path):
             if None in (key, obj):
                 return obj
+            if isinstance(obj, re.Match):
+                obj = obj.groupdict()
```

Comment:
note to self: reimplement this in new traverse_obj

Reply:
Done.